src/anim/anim.js

```diff
@@ -508,48 +498,258 @@ Object.assign(pc, function () {
         }
     });
 
+    /**
+     * @private
+     * @callback pc.AnimSetter
+     * @description Callback function for applying an updated animation value to some target.
+     * @param {[number]} value - updated animation value.
+     */
+
     /**
      * @private
      * @class
-     * @name pc.AnimController
-     * @classdesc AnimContoller stores a set of animation clips and performs blending
-     * between them. It then applies the resulting transforms to the target nodes.
-     * @description Create a new animation controller.
-     * @param {pc.GraphNode} graph - root of the scene graph to control.
+     * @name pc.AnimTarget
+     * @classdesc Stores the information required by {@link pc.AnimController} for updating a target value.
+     * @param {pc.AnimSetter} func - this function will be called when a new animation value is output by
+     * the {@link pc.AnimController}.
+     * @param {'vector'|'quaternion'} type - the type of animation data this target expects.
```

Comment:
So is a boolean or number target a vector type with 1 component?

Reply:
Yes. Presumably a boolean would be a float vec[1] with step interpolation and values limited 0.0's and 1.0's.